### PR TITLE
Drop a couple of lodash functions

### PR DESCRIPTION
**Pull request by fregante**
- Closes https://github.com/pixiebrix/pixiebrix-extension/issues/816

This only replaces a couple of simple functions, more commonly the `fromPairs`, assuming `Object.fromEntries` is identical. 

I can tell the current lodash usage is really effective so no need to replace it with slightly less convenient options, like:

- ❌ `castArray(x)` -> `Array.isArray(x) ? x : [x]`
- ❌ `uniq(x)` -> `[... new Set(X)]`

in some cases lodash usage _could_ be avoided but it's not worth having to rethink existing code.

## Current stats

<details>
<summary>Exact imports</summary>

```
❯ rg 'import \\{[^}]+\\} from "lodash"' --multiline-dotall --multiline
src/validators/generic.ts
30:import { pickBy, isEmpty } from "lodash";

src/services/baseService.ts
18:import { isEmpty } from "lodash";

src/services/useDependency.ts
26:import { castArray, head } from "lodash";

src/validators/validation.ts
25:import { uniq, mapValues, isPlainObject } from "lodash";

src/frameworks/contrib/vue.ts
23:import { pickBy, isEmpty, set } from "lodash";

src/services/factory.ts
34:import { isEmpty, castArray, uniq, compact } from "lodash";

src/services/locator.ts
31:import { sortBy, isEmpty } from "lodash";

src/frameworks/contrib/react.ts
19:import { pickBy } from "lodash";

src/pages/marketplace/MarketplacePage.tsx
21:import { sortBy } from "lodash";

src/frameworks/contrib/ember.ts
22:import { mapValues, partial, unary } from "lodash";

src/pages/marketplace/useReinstall.ts
24:import { groupBy, uniq } from "lodash";

src/frameworks/contrib/angularjs.ts
22:import { pickBy, isEmpty } from "lodash";

src/pages/marketplace/useInstall.ts
26:import { uniq } from "lodash";

src/nativeEditor/selector.ts
21:import { uniq } from "lodash";

src/nativeEditor/frameworks.ts
20:import { uniq, isEmpty } from "lodash";

src/nativeEditor/infer.ts
18:import { uniq, compact, sortBy, unary, intersection } from "lodash";

src/contentScript/notify.ts
19:import { merge } from "lodash";

src/contentScript/lifecycle.ts
32:import { groupBy } from "lodash";

src/registry/internal.ts
29:import { cloneDeep, isEmpty, isPlainObject, mapValues, pick } from "lodash";

src/registry/localRegistry.ts
19:import { sortBy, groupBy } from "lodash";

src/hooks/useRefresh.ts
21:import { stubTrue } from "lodash";

src/hooks/organization.ts
21:import { partial } from "lodash";

src/icons/IconSelector.tsx
28:import { sortBy } from "lodash";

src/devTools/ScopeSettings.tsx
22:import { castArray, mapValues } from "lodash";

src/devTools/editor/extensionPoints/base.ts
27:import { castArray, isPlainObject } from "lodash";

src/devTools/editor/extensionPoints/trigger.tsx
40:import { castArray, identity, pickBy } from "lodash";

src/devTools/editor/extensionPoints/actionPanel.tsx
32:import { castArray, identity, pickBy } from "lodash";

src/devTools/editor/hooks/useCreate.ts
24:import { uniq } from "lodash";

src/core.ts
24:import { pick } from "lodash";

src/vendors/libraryDetector/detect.ts
6:import { zip } from "lodash";

src/devTools/editor/hooks/useAvailableExtensionPoints.ts
25:import { zip } from "lodash";

src/devTools/editor/hooks/useReservedNames.ts
21:import { sortBy, uniq } from "lodash";

src/devTools/editor/hooks/useInstallState.ts
27:import { zip } from "lodash";

src/devTools/editor/components/Effect.tsx
21:import { isEqual } from "lodash";

src/devTools/editor/fields/SelectorSelectorField.tsx
20:import { compact, isEmpty, sortBy, uniqBy } from "lodash";

src/devTools/editor/extensionPoints/contextMenu.tsx
34:import { castArray, identity, pickBy } from "lodash";

src/devTools/editor/extensionPoints/menuItem.ts
42:import { castArray, identity, pickBy } from "lodash";

src/devTools/editor/ElementWizard.tsx
21:import { isEmpty, groupBy } from "lodash";

src/devTools/editor/fields/creatableAutosuggest/CreatableAutosuggest.tsx
26:import { noop } from "lodash";

src/devTools/editor/extensionPoints/panel.ts
32:import { castArray, identity, pickBy } from "lodash";

src/devTools/Locator.tsx
27:import { isEmpty } from "lodash";

src/contrib/zapier/pushOptions.tsx
24:import { compact } from "lodash";

src/contrib/automationanywhere/run.ts
20:import { mapValues } from "lodash";

src/contrib/automationanywhere/BotOptions.tsx
24:import { isEmpty, compact } from "lodash";

src/contrib/uipath/ProcessOptions.tsx
24:import { compact } from "lodash";

src/contrib/uipath/embedApp.ts
19:import { isEmpty } from "lodash";

src/contrib/uipath/LocalProcessOptions.tsx
23:import { compact } from "lodash";

src/components/fields/BlockField.tsx
22:import { castArray, isEmpty } from "lodash";

src/components/fields/TextField.tsx
20:import { uniq } from "lodash";

src/components/fields/FieldTable.tsx
22:import { isEmpty } from "lodash";

src/components/fields/BlockModal.tsx
29:import { sortBy, unary } from "lodash";

src/contrib/google/geocode.ts
18:import { isEmpty } from "lodash";

src/components/fields/ServiceModal.tsx
29:import { sortBy } from "lodash";

src/contrib/google/bigquery/query.ts
22:import { zipObject } from "lodash";

src/components/fields/TextField.stories.tsx
22:import { noop } from "lodash";

src/contrib/google/sheets/AppendSpreadsheetOptions.tsx
25:import { compact, uniq } from "lodash";

src/contrib/google/sheets/FileField.tsx
31:import { partial } from "lodash";

src/contrib/google/sheets/append.ts
27:import { unary } from "lodash";

src/components/fields/blockOptions.tsx
28:import { uniq, sortBy, isEmpty, compact } from "lodash";

src/contrib/pipedrive/readers.ts
19:import { startCase, mapValues } from "lodash";

src/components/logViewer/useLogEntries.ts
26:import { stubTrue } from "lodash";

src/components/logViewer/LogToolbar.tsx
20:import { range } from "lodash";

src/components/logViewer/LogContext.tsx
20:import { isEqual, noop } from "lodash";

src/contrib/hubspot/upsert.ts
21:import { partial } from "lodash";

src/permissions/index.ts
26:import { castArray, compact, uniq } from "lodash";

src/components/schemaTree/SchemaTree.tsx
21:import { isEmpty, sortBy } from "lodash";

src/devTools/editor/tabs/EffectTab.tsx
22:import { zip } from "lodash";

src/components/schemaTree/entries/ObjectEntry.tsx
20:import { isEmpty } from "lodash";

src/devTools/editor/tabs/ServicesTab.tsx
23:import { head } from "lodash";

src/devTools/editor/sidebar/Sidebar.tsx
21:import { sortBy } from "lodash";

src/devtools.ts
25:import { once } from "lodash";

src/devTools/editor/tabs/reader/ReaderConfig.tsx
21:import { compact, isEmpty, mapValues, partial, pick, pickBy } from "lodash";

src/utils.ts
18:import {
19:  isEmpty,
20:  mapValues,
21:  partial,
22:  partialRight,
23:  negate,
24:  identity,
25:  countBy,
26:  maxBy,
27:  entries,
28:  last,
29:  flow,
30:  head,
31:  ObjectIterator,
32:  zip,
33:  pickBy,
34:  isPlainObject,
35:} from "lodash";

src/utils/objToYaml.ts
18:import { sortBy } from "lodash";

src/utils/renderers.ts
20:import { mapKeys, once } from "lodash";

src/utils/permissions.ts
19:import { uniq } from "lodash";

src/tests/errors.test.ts
12:import { range } from "lodash";

src/extensionPoints/actionPanelExtension.ts
57:import { cloneDeep } from "lodash";

src/extensionPoints/menuItemExtension.ts
22:import { castArray, once, debounce, cloneDeep } from "lodash";

src/extensionPoints/triggerExtension.ts
39:import { castArray, cloneDeep, compact } from "lodash";

src/extensionPoints/panelExtension.ts
23:import { castArray, cloneDeep } from "lodash";

src/extensionPoints/contextMenu.ts
40:import { castArray, uniq, compact, cloneDeep, isEmpty } from "lodash";

src/extensionPoints/helpers.ts
18:import { castArray, noop, once } from "lodash";

src/script.ts
46:import { isEmpty, identity, castArray, cloneDeep } from "lodash";

src/helpers.ts
19:import { mapValues, pickBy, isPlainObject } from "lodash";

src/auth/token.ts
21:import { isEqual } from "lodash";

src/options/slices.ts
28:import { orderBy } from "lodash";

src/options/pages/MarketplacePage.tsx
20:import { compact } from "lodash";

src/baseRegistry.ts
27:import { groupBy } from "lodash";

src/options/pages/services/useServiceDefinitions.ts
23:import { sortBy } from "lodash";

src/options/pages/installed/ShareExtensionModal.tsx
20:import { compact, isEmpty } from "lodash";

src/options/pages/brickEditor/useSubmitBrick.tsx
19:import { castArray } from "lodash";

src/options/pages/brickEditor/Sharing.tsx
22:import { sortBy, uniq } from "lodash";

src/options/pages/marketplace/ConfigureBody.tsx
23:import { identity, pickBy } from "lodash";

src/options/pages/brickEditor/CodeEditor.tsx
18:import { castArray, trim, noop, isEmpty } from "lodash";

src/options/pages/marketplace/ActivateWizard.tsx
21:import { isEmpty, mapValues, uniq } from "lodash";

src/options/pages/brickEditor/Editor.tsx
31:import { sortBy } from "lodash";

src/options/pages/brickEditor/referenceTab/BrickReference.tsx
29:import { sortBy } from "lodash";

src/options/pages/brickEditor/referenceTab/BrickDetail.tsx
20:import { isEmpty } from "lodash";

src/options/pages/marketplace/PermissionsBody.tsx
19:import { uniq } from "lodash";

src/options/pages/installed/useExtensionPermissions.ts
21:import { castArray } from "lodash";

src/options/pages/workshop/WorkshopPage.tsx
47:import { orderBy, uniq, compact, sortBy, isEmpty } from "lodash";

src/options/pages/installed/exportBlueprint.ts
18:import { isEmpty } from "lodash";

src/options/pages/installed/ActiveBricksCard.tsx
24:import { groupBy } from "lodash";

src/options/pages/templates/TemplatesPage.tsx
27:import { groupBy, sortBy, noop, compact } from "lodash";

src/options/pages/installed/ExtensionRows.tsx
23:import { sortBy } from "lodash";

src/options/pages/services/ServiceEditorModal.tsx
25:import { cloneDeep } from "lodash";

src/options/pages/settings/AdvancedSettings.tsx
24:import { isEmpty } from "lodash";

src/background/logging.ts
25:import { sortBy, isEmpty } from "lodash";

src/blocks/combinators.ts
41:import { castArray, isEmpty, isPlainObject, mapValues, pickBy } from "lodash";

src/background/deployment.ts
22:import { partition, uniqBy } from "lodash";

src/background/telemetry.ts
21:import { compact, debounce, throttle, uniq } from "lodash";

src/blocks/readers/frameworkReader.ts
21:import { castArray, compact } from "lodash";

src/blocks/readers/CompositeReader.ts
20:import { mapValues, identity } from "lodash";

src/blocks/readers/factory.ts
27:import { cloneDeep } from "lodash";

src/blocks/readers/ArrayCompositeReader.ts
20:import { identity, zip } from "lodash";

src/blocks/effects/pageState.ts
21:import { merge, cloneDeep } from "lodash";

src/background/devtools/protocol.ts
33:import { isEmpty } from "lodash";

src/blocks/available.ts
19:import { castArray } from "lodash";

src/blocks/transformers/blockFactory.ts
22:import { castArray } from "lodash";

src/background/requests.ts
27:import { isEmpty } from "lodash";

src/blocks/transformers/regex.ts
21:import { unary } from "lodash";

src/blocks/transformers/parseUrl.ts
21:import { pick } from "lodash";

src/background/executor.ts
37:import { partition, zip } from "lodash";

src/blocks/errors.ts
18:import { castArray } from "lodash";

src/background/contextMenus.ts
23:import { noop } from "lodash";

src/options/pages/settings/PermissionsSettings.tsx
22:import { sortBy } from "lodash";

src/blocks/renderers/propertyTable.tsx
22:import { sortBy, isPlainObject } from "lodash";
```

</details>

<details>
<summary>functions used</summary>

```
castArray
cloneDeep
compact
countBy
debounce
entries
flow
groupBy
head
identity
intersection
isEmpty
isEqual
isPlainObject
last
mapKeys
mapValues
maxBy
merge
negate
noop
once
orderBy
partial
partialRight
partition
pick
pickBy
range
set
sortBy
startCase
stubTrue
throttle
trim
unary
uniq
uniqBy
zip
zipObject
```

</details>

### CODE REVIEW

**Comment by fregante** [comment]
If we find a way to do it via webpack, it'd be great to automagically replace some functions with their [obvious counterparts](https://twitter.com/fregante/status/1427577255790714883) at build time, without bothering the lodash zen garden.

**Review by twschiller — CHANGES_REQUESTED** [review]
The behavior of truncate is different than slice. Truncate adds ellipses or a provided omission string https://lodash.com/docs/4.17.15#truncate

I'm fine with enforcing these rules for fromPairs and reverse to make reading code easier. E.g., I didn't realize lodash's reverse also mutates the array. I don't think lodash moves the needle on performance, so my main concern would be on productivity/readability

- [x] Revert truncate changes

File: src/types.ts [hunk 1]
```diff
@@ -143,7 +142,7 @@ export abstract class ExtensionPoint<TConfig extends EmptyConfig>
     const before = this.extensions.map((x) => x.id);
 
     const updatedIds = new Set(extensions.map((x) => x.id));
-    const [, removed] = partition(this.extensions, (currentExtension) =>
+    const removed = this.extensions.filter((currentExtension) =>
```
Comment (twschiller) [inline src/types.ts:145]: Isn't this transformation wrong? https://lodash.com/docs/4.17.15#partition

The removed is what the partition returned falsey for. 

> Creates an array of elements split into two groups, the first of which contains elements predicate returns truthy for, the second of which contains elements predicate returns falsey for. The predicate is invoked with one argument

https://lodash.com/docs/4.17.15#partition

- [ ] Fix transformation of partition

**Review by twschiller — CHANGES_REQUESTED** [review]
Fix transform of partition

**Review by twschiller — APPROVED** [review]

**Pull request merged**